Source/scrollrt.cpp

```diff
@@ -946,38 +946,57 @@ static void scrollrt_draw(CelOutputBuffer out, int x, int y, int sx, int sy, int
 }
 
 /**
- * @brief Scale up the rendered part of the back buffer to take up the full view
+ * @brief Scale up the top left part of the buffer 2x.
  */
 static void Zoom(CelOutputBuffer out)
 {
-	int wdt = gnScreenWidth / 2;
-
-	int src_x = gnScreenWidth / 2 - 1;
-	int dst_x = gnScreenWidth - 1;
-
+	int viewport_width = out.w();
+	int viewport_offset_x = 0;
 	if (PANELS_COVER) {
 		if (chrflag || questlog) {
-			wdt >>= 1;
-			src_x -= wdt;
+			viewport_width -= SPANEL_WIDTH;
+			viewport_offset_x = SPANEL_WIDTH;
 		} else if (invflag || sbookflag) {
-			wdt >>= 1;
-			src_x -= wdt;
-			dst_x -= SPANEL_WIDTH;
+			viewport_width -= SPANEL_WIDTH;
 		}
 	}
 
-	BYTE *src = out.at(src_x, gnViewportHeight / 2 - 1);
-	BYTE *dst = out.at(dst_x, gnViewportHeight - 1);
+	// We round to even for the source width and height.
+	// If the width / height was odd, we copy just one extra pixel / row later on.
+	const int src_width = (viewport_width + 1) / 2;
+	const int doubleable_width = viewport_width / 2;

```

Comment:
Is `doubleable` a word? What does this mean exactly?

Reply:
I couldn't come up with a better term, but this is the width the algorithm doubles.
Then there may be a single-pixel column/row that is copied as is, in case the viewport dimension is odd.